### PR TITLE
feat: introducing chain processors

### PR DESCRIPTION
**Pull request by chr-hertel**
Idea to decouple functionality and extension points for chain - not working yet though

WDYT?

### CODE REVIEW

**Comment by OskarStark** [comment]
Sounds like a nice idea!

**Comment by chr-hertel** [comment]
works now, but need to think about it ...

File: src/OpenAI/Model/Gpt.php [hunk 1]
```diff
@@ -40,6 +40,11 @@ public function supportsToolCalling(): bool
         return true;
     }
 
+    public function supportsImageInput(): bool
+    {
+        return $this->version->supportImageInput;
+    }
```
Comment (OskarStark) [inline src/OpenAI/Model/Gpt.php:46]: The logic which belongs to image support should be extracted to a separate PR to avoid noise around the new chain processor feature

Comment (chr-hertel) [reply src/OpenAI/Model/Gpt.php:46]: True, will do so tomorrow
